pkg/resource/secret_reconciler.go

```diff
@@ -33,33 +35,31 @@ import (
 // propagated to the named resource of the same kind, assuming it exists and
 // consents to propagation.
 const (
-	AnnotationKeyPropagateToNamespace = "crossplane.io/propagate-to-namespace"
-	AnnotationKeyPropagateToName      = "crossplane.io/propagate-to-name"
-	AnnotationKeyPropagateToUID       = "crossplane.io/propagate-to-uid"
-)
-
-// Supported resources with all of these annotations consent to be fully or
-// partially propagated from the named resource of the same kind.
-const (
-	AnnotationKeyPropagateFromNamespace = "crossplane.io/propagate-from-namespace"
-	AnnotationKeyPropagateFromName      = "crossplane.io/propagate-from-name"
-	AnnotationKeyPropagateFromUID       = "crossplane.io/propagate-from-uid"
+	AnnotationKeyPropagateTo         = "to.propagate.crossplane.io/"
+	AnnotationKeyPropagateToFormat   = "to.propagate.crossplane.io/%s"
+	AnnotationKeyPropagateFrom       = "from.propagate.crossplane.io/"
+	AnnotationKeyPropagateFromFormat = "from.propagate.crossplane.io/%s"
```

Comment:
I notice you're using `fmt.Sprintf` for annotation keys, and `strings.Join` for annotation values. Perhaps we should settle on one or the other? My vote would be for `strings.Join`, given we already use `AnnotationKeyPropagateTo` and `AnnotationKeyPropagateFrom` to test for prefix matches.

Also, perhaps these should be `AnnotationKeyPrefixPropagateTo` (and `From`) to clarify that they're a prefix?